src/server/service/kv.rs

```diff
@@ -332,28 +343,28 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
         let (cb, f) = paired_future_callback();
         let res = self.gc_worker.start_collecting(req.get_max_ts().into(), cb);
 
-        let future = AndThenWith::new(res, f.map_err(Error::from))
-            .and_then(|v| {
-                let mut resp = RegisterLockObserverResponse::default();
-                if let Err(e) = v {
-                    resp.set_error(format!("{}", e));
-                }
-                sink.success(resp).map_err(Error::from)
-            })
-            .map(move |_| {
-                GRPC_MSG_HISTOGRAM_STATIC
-                    .register_lock_observer
-                    .observe(duration_to_sec(begin_instant.elapsed()))
-            })
-            .map_err(move |e| {
-                debug!("kv rpc failed";
-                    "request" => "register_lock_observer",
-                    "err" => ?e
-                );
-                GRPC_MSG_FAIL_COUNTER.register_lock_observer.inc();
-            });
+        let task = async move {
```

Comment:
I don't think it's correct. In the original implement, error is passed to response.

Reply:
fixed